core/src/test/java/org/elasticsearch/index/store/CorruptedFileIT.java

```diff
@@ -568,34 +590,21 @@ private int numShards(String... index) {
         return shardIterators.size();
     }
 
-    private Map<String, List<Path>> findFilesToCorruptForReplica() throws IOException {
-        Map<String, List<Path>> filesToNodes = new HashMap<>();
-        ClusterState state = client().admin().cluster().prepareState().get().getState();
-        Index test = state.metaData().index("test").getIndex();
-        for (ShardRouting shardRouting : state.getRoutingTable().allShards("test")) {
-            if (shardRouting.primary()) {
```

Comment:
The method name is now misleading. let me explain. I run into a situation that while shutting down the cluster we shutdown the node with a primary first and the dedicated masters managed to promote replica to primary and persist that promotion before shutting down. This meant that we ended up corrupting the only valid copy of the data and the cluster will not become green. I *think* this didn't happened before because stopping and starting nodes in the same order and thus the first node started (which becomes a master)  always used the cluster state from before the restarted where the primaries have not been changed yet. This was trappy imo so I changed it to allow primaries to first be elected POST restarted and then go and corrupt all unused copies. I will change the method name to reflect that.